Scripts/ShaderReloadSoakTest.bv.lua

```diff
@@ -12,27 +12,36 @@
 -- WARNING: This is a soak test, do not add this test to the fully automated
 -- test suites.
 
-OpenSample('RPI/ShaderReloadSoakTest')
+OpenSample('RPI/ShaderReloadTest')
 ResizeViewport(500, 500)
-IdleSeconds(2)
 
-function TestButton(buttonName, shaderReloadWaitSeconds, pixelCaptureWaitSeconds)
+
+function TestButton(buttonName)
+    AssetTracking_Start()
+    AssetTracking_ExpectAsset("shaders/shaderreloadtest/temp/fullscreen.shader")
     SetImguiValue(buttonName, true)
-    IdleSeconds(shaderReloadWaitSeconds) -- Idle for a bit to give time for the shader to reload
+    AssetTracking_IdleUntilExpectedAssetsFinish(10)
+    -- Even though the shader has been recompiled, it takes a few frames
+    -- for the notification to bubble up and get the screen refreshed.
+    IdleSeconds(0.25)
+
     SetShowImGui(false)
     SetImguiValue('Check color', true)
-    IdleSeconds(pixelCaptureWaitSeconds)
+
+    -- Need a few frames to capture the screen and compare expected color.
+    IdleSeconds(0.25)
```

Comment:
This seems small, but when you do it many times it adds up (3.75s in this case). You could probably leverage ScriptRunnerRequests::PauseScript/ResumeScript in the component to pause while the screenshot is being taken.

Actually, why not use the screenshot comparison process that we use in other tests? Then you could remove the Check Color button, and all the bespoke frame capture code you have in the component.

Reply:
Will do that after reinvent. But I think We need other APIs for tests that don;t require full screen capture comparison.